src/main/java/rosmop/codegen/HeaderGenerator.java

```diff
@@ -20,12 +20,16 @@
  */
 public class HeaderGenerator {
 
-	protected final static SourcePrinter printer = new SourcePrinter();
+	protected static SourcePrinter printer = new SourcePrinter();
 	static boolean hasInit = false;
 	static HashMap<String, ArrayList<ROSEvent>> addedTopics = 
 			new HashMap<String, ArrayList<ROSEvent>>();
 
 
+	public static void reset() {
+		printer = new SourcePrinter();
+		addedTopics.clear();
+	}
```

Comment:
This is OK for now, but a better pattern is to change all methods of HeaderGenerator to be non-static, and have a single static method `HeaderGenerator.getInstance()` to access the global instance when needed. This way if new members are added to `HeaderGenerator` we will not need to remember to reset them too.